audit-checklists/Chainlink-CCIP.md

```diff
@@ -0,0 +1,104 @@
+    Client.EVMExtraArgsV2({
+        gasLimit: 200_000,
+        allowOutOfOrderExecution: true
+    })
+)
+```
+
+- **`gasLimit`**: Specifies the maximum amount of gas that CCIP can consume to execute `ccipReceive()` on the contract located on the destination blockchain. It is the main factor in determining the fee to send a message.   
+  - If `gasLimit` is not provided, the default value is `200,000`. Ensure that `ccipReceive()` does not require more gas than this default.
+  - Unspent gas is **not refunded**.
+- **`allowOutOfOrderExecution`**: Controls the execution order of your messages on the destination blockchain. This parameter is available only on lanes where the `Out of Order Execution` property is set to *Optional* or *Required*.  
+  - When `allowOutOfOrderExecution` is **Optional**: You can set it to either `true` or `false`.  
+  - When `allowOutOfOrderExecution` is **Required**: You must set it to `true`. This acknowledges that messages may be executed out of order. If set to `false`, the message will revert and will not be processed.
+
+> **Note**: It is recommended to use mutable `extraArgs` (not hardcoded) in production deployments.
+
+
+### CCIP Rate Limits
+Rate limits consist of a maximum capacity and a refill rate, which determines how quickly the maximum capacity is restored after a token transfer consumes some or all of the available capacity.
+
+- **[Token Pool Rate Limit](https://docs.chain.link/ccip/architecture#token-pool-rate-limit)**: For each supported token on every individual [lane](https://docs.chain.link/ccip/concepts#lane), this rate limit manages the total number of tokens that can be transferred within a specified time frame. This limit is independent of the token's USD value.  
+- **[Aggregate Rate Limit](https://docs.chain.link/ccip/architecture#aggregate-rate-limit)**: Each lane also has an aggregate rate limit that caps the total USD value of transfers across all supported tokens on that lane.
+
+
+## Token Decimal Handling
+When tokens move between blockchains with different decimal places, rounding may occur. This rounding can impact small amounts of tokens during cross-chain transfers.  
+
+**Example**:  
+Token precision on Chain A is 18 decimals, and on Chain B is 9 decimals.  
+```
+• Sent from A: 1.123456789123456789  
+• Received on B: 1.123456789  
+
+• Lost: 0.000000000123456789  
+```
+
+
+## Manual Execution
+In certain exceptional conditions, users might need to manually execute the transaction on the destination blockchain. 
+These conditions include:
+
+- **Unhandled exceptions**: Logical errors in the receiver contract.  
+- **Gas limit exceeded for token pools**: If the combined execution of the required functions (`balanceOf` checks and [releaseOrMint](https://github.com/smartcontractkit/ccip/blob/release/contracts-ccip-1.5.1/contracts/src/v0.8/ccip/pools/BurnMintTokenPoolAbstract.sol#L36)) surpasses the default gas limit of 90,000 on the destination blockchain.
```

Comment:
Can you point to the exact commit and not the branch? Branch can get deleted so it will become a dead link.